### ISSUE TITLE
Create entries without oauth?

### ISSUE THREAD

**Issue by pmudahar** [mcohen01/node-quickbooks#88]
I'm not trying to create a quickbooks apps for other people to use.  I am trying to connect a web application with the company quickbooks so that certain events from the website trigger entries in quickbooks.

For this use case, we do not want to login to quickbooks from the web app.  Are there keys associated with the qb account that would enable us to create journal entries from the node backend of the web application?

In other words, can the app obtain a auth token without actually requiring the user to login? As long as the correct credentials are presented?

I hope that was clear - please let me know if you have any questions.

Thanks in advance!
Paul

**Comment by mcohen01** [comment]
The OAuth tokens are valid for 6 months, so you can just get the keys and then store them somewhere secure that your webapp can read from. Every 6 months you'll have to refresh them.

**Comment by mLuby** [comment]
For OAuth 2.0 (which AFAIK is the only type available to new dev accounts now), the token [expires in 1 hour](https://developer.intuit.com/docs/00_quickbooks_online/2_build/10_authentication_and_authorization/10_oauth_2.0#/Handling_the_response_2). Dealing with OAuth and token refreshing is such a waste of dev time for single user server applications. 😖 Does anyone know a way around this?

**Comment by octalsolutions** [comment]
@pmudahar Did you found any solution on this? same as you I need to sync my inventory with quickbooks inventory.

**Comment by pmudahar** [comment]
sorry @octalsolutions  i didnt end up implementing it.  but if you figure out a solution, please post it here.
